### PR TITLE
fixes bug where notebooks / revisions cannot be deleted from server

### LINKED ISSUE #1773: "delete notebook" on server appears to be 404ing now
This was likely introduced by a recent change. I think it'll be an easy fix.

### PR DESCRIPTION
**Pull request by hamilton**
fixes #1773, which was likely introduced by the file API PR. This fixes a couple of small aspects of file uploading from the server, but is not meant to refactor the functionality in any substantial way.

### Pull Request checklist

- [x] **Documentation**: If this feature has or requires documentation, the relevant docs have been updated.
- [x] **Changelog**: This PR updates the [changelog](../CHANGELOG.md) with any user-visible changes.
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not

### CODE REVIEW

File: src/server/components/delete-modal.jsx [hunk 1]
```diff
@@ -15,7 +14,8 @@ export default class DeleteModal extends React.Component {
   deleteObject() {
     // add delete here
 
-    deleteFileOnServer(this.props.elementID)
+    this.props
+      .deleteFunction(this.props.elementID)
```
Comment (hamilton) [inline src/server/components/delete-modal.jsx]: by moving to `this.props.deleteFunction` we can test the actual deleting function independently, and we don't have to pass around boilerplate url strings

File: src/shared/utils/file-operations.js [hunk 2]
```diff
@@ -95,23 +91,29 @@ export function saveFileToServer(
...
     if (r.status === 500) throw new Error(r.statusText);
     return r.json();
   });
 }
 
-export function deleteFileOnServer(fileID) {
-  return fetchWithCSRFTokenAndJSONContent(`/api/v1/files/${fileID}/`, {
-    method: "DELETE"
-  }).then(r => {
+export function makeDeleteRequest(url) {
```
Comment (hamilton) [inline src/shared/utils/file-operations.js]: most delete requests have the exact same structure, with the difference being the URL in which we suggest to delete. This function should not ever be used – rather, I would suggest following the approach used in eg `deleteNotebookOnServer` below.

Comment (bcolloran) [reply src/shared/utils/file-operations.js]: if it should not be used, should we not export it? or if we need to export it for testing, should we call it something like `makeDeleteRequest_private` and adopt a convention around not using thusly named vars outside of the file that defines them?

File: src/server/components/notebook-actions-menu.jsx [hunk 3]
```diff
@@ -64,21 +64,19 @@ export default class NotebookActionsMenu extends React.Component {
...
-      .then(response => response.json())
-      .then(response => {
-        if (this.props.onUploadFile) this.props.onUploadFile(response);
-        this.hideUploadFileConfirmationModal();
-      });
+  async updateFile() {
+    const response = await uploadFile(
+      this.state.newFile,
+      this.state.oldFile.id
+    ).then(r => r.json());
```
Comment (bcolloran) [inline src/server/components/notebook-actions-menu.jsx]: here and above we could kill off another `then` with 

```
 const data = await response.json();
```

File: src/server/components/file-actions-menu.jsx [hunk 4]
```diff
@@ -40,11 +41,11 @@ export default class FileActionsMenu extends React.Component {
           visible={this.state.deleteModalVisible}
           onClose={this.hideDeleteModal}
           title={`delete the file  "${this.props.filename}"?`}
+          deleteFunction={deleteFileOnServer}
```
[the remark this answers is not in the document]
Comment (hamilton) [reply src/server/components/file-actions-menu.jsx]: yep, I filed this before that merged but will rebase accordingly

File: src/shared/utils/file-operations.js [hunk 5]
```diff
@@ -34,17 +34,13 @@ export function selectFileAndFormatMetadata(notebookID) {
   });
 }
 
-export function uploadFile(formData) {
-  return fetchWithCSRFToken("/api/v1/files/", {
+export function uploadFile(formData, fileID = undefined) {
+  // if fileID not defined, this will upload the file, or return an error if
+  // the filename in formData is already in the files table.
+  // if fileID is provided, will attempt to update the entry in the files table.
+  return fetchWithCSRFToken(`/api/v1/files/${fileID ? `${fileID}/` : ""}`, {
```
[the remark this answers is not in the document]
Comment (hamilton) [reply src/shared/utils/file-operations.js]: that won't really work without the additional trailing slash so I'll just refactor this bit

**Comment by hamilton** [comment]
@bcolloran thanks for the review. Updated + merging now.

**Pull request merged**